docs/iris/src/conf.py

```diff
@@ -64,7 +64,6 @@
               'sphinx.ext.imgmath',
               'sphinx.ext.intersphinx',
               'matplotlib.sphinxext.mathmpl',
-              'matplotlib.sphinxext.only_directives',
```

Comment:
After a little bit of investigation, I happened across https://github.com/matplotlib/matplotlib/pull/11295...

In a nutshell, ditch the `matplotlib` custom directives, which have been retired. The impact is simply to switch from `matplotlib` to the appropriate `sphinx` directives i.e.,

- replace `.. htmlonly::` -> `.. only:: html`, and
- replace `.. latexonly::` -> `.. only:: latex`